include/RED4ext/DynArray.hpp

```diff
@@ -188,14 +188,25 @@ struct DynArray
         if (capacity >= aCount)
             return;
 
+        auto newCapacity = CalculateGrowth(aCount);
+        Resize(newCapacity);
+    }
+
+    void ShrinkToSize()
+    {
+        if (capacity > size)
+            Resize(size);
+    }
+
+    void Resize(uint32_t aNewCapacity)
```

Comment:
This might create undefined behavior since elements are not constructed and destructed when the size change.

Reply:
I feel like there's some confusion. The new method shouldn't change the size, but only calacity, i.e. it's functionality is limited to "reserve" and "shrink".
If "resize" method is actually needed it shouldn't be mixed up with capacity control.

Reply:
Thjat's a good one, I also suggest making this `private` then.

Reply:
`SetCapacity`

Reply:
> What exactly do you want me to make private? `ShrinkToSize()`?

`SetCapacity` can be private, since you already have `ShrinkToSize` and `Resize`.

Reply:
Ah, nvm I see now that `SetCapacity` is private.

Reply:
Resize was already renamed to be `SetCapacity` and made private